libs/wingc/src/files.rs

```diff
@@ -38,7 +38,28 @@ impl Error for FilesError {}
 
 #[derive(Default)]
 pub struct Files {
-	data: HashMap<Utf8PathBuf, String>,
+	pub data: HashMap<Utf8PathBuf, FileData>,
+}
+
+pub struct FileData {
```

Comment:
Still iffy about this

Reply:
Yeah this does feel a bit weird. I'm not a fan of exposing `data` as a public field, since that means `Files` is no longer well encapsulated (anyone can mutate the hashmap as they please, regardless of what contracts its methods are trying to enforce)

Reply:
Yea I changed this back to private and put all the work in the impl.